lib/static/modules/reducers/suites-page.ts

```diff
@@ -1,15 +1,22 @@
-import actionNames from '../action-names';
 import {State} from '@/static/new-ui/types/store';
-import {Action} from '@/static/modules/actions/types';
+import {SuitesPageAction} from '@/static/modules/actions/suites-page';
+import actionNames from '@/static/modules/action-names';
 
-export type SuitesPageSetCurrentSuiteAction = Action<typeof actionNames.SUITES_PAGE_SET_CURRENT_SUITE, {
-    suiteId: string;
-}>;
-
-export default (state: State, action: SuitesPageSetCurrentSuiteAction): State => {
+export default (state: State, action: SuitesPageAction): State => {
     switch (action.type) {
         case actionNames.SUITES_PAGE_SET_CURRENT_SUITE:
             return {...state, app: {...state.app, currentSuiteId: action.payload.suiteId}};
+        case actionNames.SUITES_PAGE_SET_SECTION_EXPANDED: {
+            const newState = Object.assign({}, state);
+            const newExpandedSectionsById = Object.assign({}, state.ui.suitesPage.expandedSectionsById, {
+                [action.payload.sectionId]: action.payload.isExpanded
+            });
+            newState.ui = Object.assign({}, state.ui, {
+                suitesPage: Object.assign({}, state.ui.suitesPage, {expandedSectionsById: newExpandedSectionsById})
+            });
+
+            return newState;
```

Comment:
jfiy: you could do this like this:
```ts
import {set} from 'lodash';
import {applyStateUpdate} from '../utils';

const diff = {}; // difference between current state and the new one
set(
    diff,
    ["ui", "suitesPage", "expandedSectionsById", action.payload.sectionId], 
    action.payload.isExpanded
);

return applyStateUpdate(state, diff);
```

Or like this:
```ts
const diff = set(
    {},
    ["ui", "suitesPage", "expandedSectionsById", action.payload.sectionId], 
    action.payload.isExpanded
);

return applyStateUpdate(state, diff);
```

You can read more about `applyStateUpdate` and other redux helpers at https://github.com/gemini-testing/html-reporter/blob/master/lib/static/modules/utils/state.js#L3-L23

Reply:
Yeah, thanks, that's much more convenient. Ideally this should be handled by Immer of course =)